dashboard/app/models/course.rb

```diff
@@ -18,6 +18,32 @@ class Course < ApplicationRecord
   has_one :plc_course, class_name: 'Plc::Course'
   has_many :course_scripts, -> {order('position ASC')}
 
+  validates :name,
+    presence: true,
+    uniqueness: {case_sensitive: false}
+
+  class CourseNameValidator < ActiveModel::Validator
+    def validate(course)
+      return if course.plc_course
+
+      unless /\A[a-z0-9\-]+\z/ =~ course.name
+        course.errors[:base] << 'can only contain lowercase letters, numbers and dashes'
+      end
+    end
+  end
+
+  validates_with CourseNameValidator
```

Comment:
Also for custom validators that are only used in a single, it's simpler to just define a [method](http://guides.rubyonrails.org/active_record_validations.html#custom-methods) ([example](https://github.com/code-dot-org/code-dot-org/blob/a56c65851abf3006d2ccb471bf166b6899eae89c/dashboard/app/models/pd/workshop.rb#L152)). These `ActiveModel::Validator` classes are typically used for validations shared across models, for [example](https://github.com/code-dot-org/code-dot-org/blob/f95e7f3a4fe0ac8d4610f80cb9e9b6d756600dc3/dashboard/lib/no_utf8mb4_validator.rb).